### ISSUE TITLE
Set roundness (border radius) of a rectangle?

### ISSUE THREAD

**Issue by borekb** [excalidraw/excalidraw#3600]
I sometimes use rectangles to indicate groups of objects, like [this](https://excalidraw.com/#json=4561586521899008,W2tgbjD760bLcR9ZbI1__Q):

<img width="1047" alt="Screen Shot 2021-05-19 at 15 38 07" src="https://user-images.githubusercontent.com/101152/118822309-55541880-b8b8-11eb-9599-929414968936.png">

I wish the rounded corners could be more subtle for these rectangles – like setting border-radius in CSS or doing this in Google Docs:

https://user-images.githubusercontent.com/101152/118822965-e6c38a80-b8b8-11eb-8d97-5b8c5a912344.mov

_I'm not sure if I missed some previous feature requests / discussions around this but I tried searching for "rounded corners", "roundness", "border radius" etc._

**Comment by dai-shi** [comment]
As a user, I wanted this feature several times.

**Comment by tsheeran1** [comment]
Upvote.  And I would like this for equilateral polygons from triangles through hexagons.

**Comment by dwelle** [comment]
@tsheeran1 when you say triangles/hexagons, you mean when you draw those shapes manually with the line tool, correct?

**Comment by kmindspark** [comment]
I'd also really like this feature

**Comment by LowLifeArcade** [comment]
I love this app and this would make it so much more useful in many use cases where I have boxes of different scales

**Comment by FraserMuir** [comment]
Would love this feature too! I can help contribute if need be. I think MVP is just an adjustable knob for the pre-set quadrilaterals, so the rectangle and diamond tool - although I would assume rectangles would be the most common usage by far, so potentially could even just start with that

**Comment by dwelle** [comment]
Two ways to do this. A UI input-slider, or display radius handles in the rectangles on canvas ala Figma:

![figma_radius](https://user-images.githubusercontent.com/5153846/127653828-45593f69-df6a-4c42-bc32-ae428627983a.gif)

Both ways add complexity the UI. I'd be more for the latter (radius handles).

We'll need to consider this.

**Comment by FraserMuir** [comment]
@dwelle Yes I'd agree not cluttering the ui should be a priority - both on-canvas and in the sidebar tooltray. I was envisioning a top-left positioned handle similar to the one shown by the OP from Google Docs, that are only visible when a given rectangle has already had the border-radius toggle activated. This would mean we'd keep the ui for regular (non-rounded) rectangles as-is.

I'd argue the 4 individual corner handles (and size measurement tooltips) from Figma are overly complex for the simplicity of excalidraw. And, applying the same logic, I'd agree with you that we shouldn't need a UI input slider in the sidebar tooltray, and that just the on-canvas handle should suffice

**Comment by jai-dewani** [comment]
If we don't want to clutter the UI, how about adding a slider in the side bar just the one that exists for opacity 

![image](https://user-images.githubusercontent.com/33520257/145413702-c5099d21-4590-4fb6-8e72-bfc6a4cf6ad1.png)

**Comment by dwelle** [comment]
First step would be to try and come up with a better default, likely adaptive (depending on shape size). One of our principles at Excalidraw is reducing as much UI buttons/toggles as possible: simplicity is key.

**Comment by jai-dewani** [comment]
If I understand correctly, having a strong curve in a small size and a small curve in a big-sized shape will be suitable.  
I'll try to play with the code changes I did for the introducing curve in diamond to understand how we can do this with a C type curve.  As of now the starting point of the curve is decided by the distance between the two corners

**Comment by lorthirk** [comment]
No news about this?

**Comment by burrijw** [comment]
+1 to this. 🙏🏻

**Comment by changer0** [comment]
+1 +1 +1

**Comment by najibghadri** [comment]
Can we like make a Chrome extension? Bc this repo is really not maintained at all let alone ship new features...

**Comment by whamsicore** [comment]
Yes!

**Comment by Davidos533** [comment]
Need this !

**Comment by PHHENS** [comment]
+1 Please allow fixed size rounded corner instead of proportional.

**Comment by georgezouq** [comment]
+1!

**Comment by m0rph03nix** [comment]
+1

**Comment by shreykx** [comment]
+1

**Comment by McVincient** [comment]
+1

**Comment by H0wAway** [comment]
great idea
